cmd/gcp-controller-manager/main.go

```diff
@@ -120,10 +123,27 @@ func main() {
 		klog.Exitf("failed loading GCP config: %v", err)
 	}
 
+	metricsMux := http.NewServeMux()
```

Comment:
All this stuff is fine and makes sense, but also kinda messy. What are the chances a 3rd handler comes along that wants to serve on 8089? E.g. /rpcz, /varz? Obviously we don't want a giant web of N! combinations of handlers. Maybe generalize this into a set or map or something. 

If you don't want to worry about that for now its no biggie.

Reply:
Yeah, good point. I changed `--healthz-port` to just `--port`, there's no reason to support per-endpoint ports. `--metrics-port` is now deprecated but supported for backwards-compatibility. It will only be used if explicitly set.